### PR TITLE
parachain backing pallet

### PR DESCRIPTION
**Pull request by xiaoch05**
Add backing pallet for `substrate<>substrate parachain` native token bridge.
For example, for `Crab <> Crab Parachain`. The native token on Crab-Chain is called Crab, we can't mint any native tokens on Crab-Parachain except the way to transfer them from Crab-Chain to Crab-Parachain by this token bridge.
The bridge needs some tokens on parachain to start. So we need pre-mint some tokens on parachain, and lock the same amount into this module on the source chain without bridge.
The decimals on Crab chain is 9 while 18 on parachain. It need to convert these different decimals between two chains.

### CODE REVIEW

**Comment by AsceticBear** [comment]
CI RED @xiaoch05

**Comment by AurevoirXavier** [comment]
> CI RED @xiaoch05

https://github.com/darwinia-network/darwinia-common/pull/1249

File: frame/wormhole/backing/parachain/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+			// this pallet account as the submitter of the remote message
+			// we need to transfer fee from user to this account to pay the bridge fee
+			T::RingCurrency::transfer(&user, &Self::pallet_account_id(), total_cost, KeepAlive)?;
+
+			let payload = T::OutboundPayloadCreator::create(
+				CallOrigin::SourceAccount(Self::pallet_account_id()),
+				spec_version,
+				weight,
+				IssuingCall::<T>::ParachainIssuingPalletIssueFromRemote(value, recipient.clone()),
```
Comment (HackFisher) [inline frame/wormhole/backing/parachain/src/lib.rs:277]: Be care about the decimal conversation here.

**Comment by HackFisher** [comment]
Companion PR on darwinia-parachain: https://github.com/darwinia-network/darwinia-parachain/pull/85

File: frame/wormhole/backing/parachain/src/lib.rs (hunk 1 above)
Comment (HackFisher) [reply frame/wormhole/backing/parachain/src/lib.rs:277]: Handled here:

https://github.com/darwinia-network/darwinia-parachain/pull/85/files#diff-cd4bcbd2cbdeec4c0771723670222324251a14c1d277112929935c99d212e375R216-R218

File: Cargo.toml [hunk 2]
```diff
@@ -37,6 +37,7 @@ members = [
 	"frame/wormhole/backing/ethereum",
 	"frame/wormhole/backing/s2s",
 	"frame/wormhole/backing/tron",
+	"frame/wormhole/backing/parachain",
```
Comment (AsceticBear) [inline Cargo.toml:40]: Remember to sort this item

File: frame/wormhole/backing/parachain/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	# paritytech
+	"bp-messages/std",
+	"bp-message-dispatch/std",
+	"bp-runtime/std",
+	"frame-benchmarking/std",
+	"frame-support/std",
+	"frame-system/std",
+	"sp-runtime/std",
+	"sp-std/std",
+	# frontier
```
Comment (AsceticBear) [inline frame/wormhole/backing/parachain/Cargo.toml:49]: Remove this line

**Review by AsceticBear — COMMENTED** [review]
Suggest adding some unit tests like other pallets to ensure core functionality is always okay. Avoid unexpected breaks in the future iteration.

**Comment by xiaoch05** [comment]
> Suggest adding some unit tests like other pallets to ensure core functionality is always okay. Avoid unexpected breaks in the future iteration. Will this pallet be rewritten with js sdk?

Yes we'll remove this pallet when the sdk is ready. I want to add the unit test case later after this pr. It has been tested by e2e test on local environment.

**Review by HackFisher — APPROVED** [review]

**Comment by AurevoirXavier** [comment]
Please link the labels/release to this PR.

**Pull request merged**